### ISSUE TITLE
Errors creating Sorghum and Cotton cultivars in Classic version 7.10 r4220

### ISSUE THREAD

**Issue by edvlmaas** [APSIMInitiative/APSIMClassic#2197]
I am running a simulation with a cotton-sorghum rotation with management driven from the Operations Schedule. I was using cultivar M35-1 with no errors, but yield was too high. It was the lowest-yielding of all the available pre-defined cultivars, so I decided to make my own.

I followed the instructions here: https://www.apsim.info/documentation/model-documentation/crop-module-documentation/adding-new-cultivars-to-an-existing-crop-module/#:~:text=To%20make%20a%20new%20cultivar%2C%20copy%20one%20that,must%20preserve%20the%20XML%20structure%20of%20the%20document.

I made a local copy of the system Sorghum.xml file, dropped the .ini onto my sorghum model and set the file name to my local copy. I made no changes at all to the local xml file, still used the M35-1 cultivar, ran the simulation, and got this:

      - reading constants for plant(sorghum) - m35-1
14 May 2004(Day of year=135): 
     !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
                      APSIM  Fatal  Error
                      -------------------
     Too many values to convert from vector to scalar
     Component name: paddock.sorghum
     
     !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

I also went ahead and tried making a new cultivar by copying and pasting the M35-1 specification in my local xml and renaming it to "lowyield". Whether I then run that with any parameter changes or not, I get the same error as above:

      - reading constants for plant(sorghum) - lowyield
14 May 2004(Day of year=135): 
     !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
                      APSIM  Fatal  Error
                      -------------------
     Too many values to convert from vector to scalar
     Component name: paddock.sorghum
     
     !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

If I try making a local copy of Cotton.xml, putting a .ini on the cotton module and setting it to the local xml, having made no other changes, I get this:
   <...>
   Component         "SurfaceOrganicMatter" = %apsim%\Model\SurfaceOM.dll
   Component                "Soil Nitrogen" = %apsim%\Model\SoilN.dll
   Component                      "sorghum" = %apsim%\Model\Sorghum.dll
   Component                     "Ryegrass" = %apsim%\Model\AgPasture.dll
   Component                       "cotton" = %apsim%\Model\Cotton.dll
     !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
                      APSIM  Fatal  Error
                      -------------------
     Exception has been thrown by the target of an invocation.
     Component name: paddock
     !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by peter-devoil** [comment]
Hi 

All I can suggest is using a [diff tool](https://en.wikipedia.org/wiki/Meld_(software)) to check your modifications; and to be sure that you've copied the Sorghum.xml from your running copy and not the repository. Once that's established, then add changes one line at a time to discover which change is breaking it.

WRT cotton - it uses its own idea of an ini file which is bundled inside the dll. [Here's](https://github.com/APSIMInitiative/APSIMClassic/issues/2031#issuecomment-815583608) how to change the parameters.

**Comment by edvlmaas** [comment]
Thank you Peter! I discovered that when I had last upgraded APSIM, it installed in a different location - I'm guessing the first time I installed it, I allowed it to be available for all users (Program Files (x86)), where the upgrade I guess I'd decided it was just for my profile (.../AppData/Local/Programs). The program version where I'd gotten the xmls from was the older one. I just grabbed the sorghum.xml from the most recent APSIM version, made a local copy with one small change, and it ran without error. Thank you!

I also followed the instructions in the other post for cotton and was able to create a cultivar and run the simulation without error. 

Many thanks for your help!
